### PR TITLE
Bring consistency among sizes

### PR DESCRIPTION
**Pull request by justinvasel**
* **What does this PR do?**

There are instances in the code that refer to lengths or sizes of strings and
buffers and the like, but they are not typed in a consistent manner. The purpose
of this commit is to provide some consistency to these scenarios by changing
their type from `unsigned int`, `int`, `unsigned long`, etc. to `size_t`.

In general, these situations were identified by globally searching the code
for patterns along the lines of:
`(unsigned|unsigned int|int|long) (len|blen|olen|length|l|size)`
and changing the type to `size_t` on a case-by-case basis.

The following situations were explicitly ignored:

- autosetup/jimsh0.c, because it appears to be third-party code

- inputs to some function calls relating to ssl and sasl, because they don't take
`size_t` as inputs and recasting in the function call didn't seem worth the
increased messiness

- for loop indices, because in some cases unsigned indices can breed bugs

This commit takes care of most obvious cases. There may certainly be subtler
ones that were missed in this pass.

* **Are there points in the code the reviewer needs to double check?**

Sanity checks are always welcome, but no specific review is requested.

* **What are the relevant issue numbers?**

Resolves: #928

### CODE REVIEW

**Comment by gahr** [comment]
Before I take a close look, I’d like to bring to your attention the fact that in some (many) cases signed integers are currently checked for > 0. This won’t work for size_t. So, have you also checked that your changes take that into account?

**Comment by justinvasel** [comment]
For signed integer to `size_t` conversions, I double checked that those variables would not potentially hold a negative value. Is that what you mean?

For example: some functions return the size of an input or -1 if unsuccessful. I would avoid switching to `size_t` in such a case.

File: conn/ssl.c [hunk 1]
```diff
@@ -740,7 +740,7 @@ static int tls_close(struct Connection *conn)
 static bool check_certificate_cache(X509 *peercert)
 {
   unsigned char peermd[EVP_MAX_MD_SIZE];
-  unsigned int peermdlen;
+  size_t peermdlen;
```
Comment (flatcap) [inline conn/ssl.c]: `X509_digest()` is expecting `unsigned int *`
When might `size_t` and `unsigned int` differ in size?

File: conn/ssl.c [hunk 2]
```diff
@@ -769,7 +769,7 @@ static bool check_certificate_cache(X509 *peercert)
 static int check_certificate_file(X509 *peercert)
 {
   unsigned char peermd[EVP_MAX_MD_SIZE];
-  unsigned int peermdlen;
+  size_t peermdlen;
```
Comment (flatcap) [inline conn/ssl.c]: `X509_digest()` again

File: pop.c [hunk 3]
```diff
@@ -83,7 +83,7 @@ static int pop_read_header(struct PopData *pop_data, struct Header *h)
 {
   FILE *f = NULL;
   int rc, index;
-  long length;
+  size_t length;
```
Comment (flatcap) [inline pop.c]: If we're using `size_t` here, we probably want to use `%zd` in the `sscanf()` below.

File: compress.c [hunk 4]
```diff
@@ -684,7 +684,7 @@ static int comp_check_mailbox(struct Context *ctx, int *index_hint)
   if (!ops)
     return -1;
 
-  int size = get_size(ctx->realpath);
+  size_t size = get_size(ctx->realpath);
```
[the remark this answers is not in the document]
Comment (justinvasel) [reply compress.c]: This is a good point. The size returned by `get_size()` comes from the `stat` struct, which uses the signed `off_t` type. I've reverted this back to `int`.

File: conn/ssl.c (hunk 1 above)
Comment (justinvasel) [reply conn/ssl.c]: As far as I can tell, there seems to be no risk in going from an `unsigned int` to `size_t`. Going the other way would be riskier as `unsigned int` is more platform-dependent in its size.

File: pop.c (hunk 3 above)
Comment (justinvasel) [reply pop.c]: Good catch. I changed it to `%zu` since `size_t` is inherently unsigned.

**Pull request merged**